otcextensions/tests/functional/sdk/dws/v1/test_tag.py

```diff
@@ -0,0 +1,86 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import os
+from openstack import _log
+from otcextensions.tests.functional import base
+
+_logger = _log.setup_logging('openstack')
+
+
+class TestClusterTags(base.BaseFunctionalTest):
+    """Functional tests for DWS cluster tags."""
+
+    def setUp(self):
+        """Prepare resources for testing."""
+        super(TestClusterTags, self).setUp()
+        self.cluster_id = os.getenv("OS_DWS_CLUSTER_ID")
+        if not self.cluster_id:
+            TestClusterTags.skipTest(
+                self, 'OS_DWS_CLUSTER_ID necessary for this test'
+            )
+        self.client = self.conn.dws
+        self.created_tags = []
+
+    def tearDown(self):
+        """Clean up resources after testing."""
+        for tag_key in self.created_tags:
+            self.client.delete_cluster_tag(self.cluster_id, tag_key)
+        self.created_tags.clear()
+        super(TestClusterTags, self).tearDown()
+
+    def test_list_cluster_tags(self):
+        """Test listing all tags of a cluster."""
+        tags = list(self.client.list_cluster_tags(self.cluster_id))

```

Comment:
you need to create instance and check that list returns more or equal 1, this test checks nothing

Reply:
Updated the test